backend/app/repository/gitlab_provider.py

```
@@ -657,4 +657,110 @@ async def _get_all_repositories_from_cache(
+        # Get API base URL based on git domain
+        api_base_url = self._get_api_base_url(git_domain)
+
+        try:
+            headers = {
+                "Authorization": f"Bearer {git_token}",
+                "Accept": "application/json"
+            }
+
+            # Get repository ID first (GitLab API uses project ID)
+            encoded_repo_name = requests.utils.quote(repo_name, safe='')
+            repo_response = requests.get(
```

Ruff (0.14.4)

Probable use of `requests` call without timeout

(S113)